proposal/proposal.tex

```diff
@@ -0,0 +1,136 @@
+\documentclass[a4paper,12pt]{article}
+
+\usepackage{geometry}
+\usepackage{hyperref}
+\geometry{margin=1.5in}
+\usepackage{enumitem}
+
+\title{Optimising \lq{matrix multiplication\rq} calculations in the tropical semiring}
+\author{Bryan Abi Karam}
+\date{\today}
+\begin{document}
+
+\maketitle
+
+\section{Introduction}
+Instruction-level parallelism is a crucial aspect of
+high performance processor design. There have been many
+previous studies that compute upper bounds on the amount
+of instruction-level parallelism in programs using mathematical
+models of idealised program execution.
+
+Specifically, one upper bound for parallelism is the number
+of dynamic instructions in a program’s execution divided by
+the length of the longest chain of dependent instructions in
+that program (the critical path). Certainly no real machine
+that takes at least one cycle per instruction can ever execute
+the program faster than the time taken to execute this critical
+path sequentially.
+
+Computing the length of this critical path can therefore be
+expressed as finding the longest path in a directed acyclic
+graph representing dependencies between dynamic instructions.
+
+Tropical analysis is a branch of pure mathematics using
+the tropical semiring (also known as the plus-min/max algebra).
+Notably, finding the longest path in a directed acyclic graph can
+be represented as a series of \lq{matrix multiplications\rq} in the
+tropical semiring.
+
+In this project I will create a library for performing such matrix
+calculations efficiently, considering aspects such as vectorisation,
+tiling, and representation sparsity, and work with both CPU and
+GPU-powered compute.
+
+\section{The project}
+
+\subsection{Project core}
+First, a CPU-based library for performing individual matrix computations in the tropical
+semiring should be implemented, using a dense representation of matrices.
+Techniques such as vectorisation and tiling should be considered for performance gains.
+
+Matrix multiplication expressions will be generated from a program
+execution's dependency structure as a graph, with one node per register per dynamic instruction,
+organised as a column of nodes per dynamic instruction.
```

Comment:
The column-based representation is not super obvious, especially given the property that each path connects to the start and end column, so maybe add e.g. "in the representation proposed by Chadwick [citation]".